unified-codes/tools/scripts/dgraph/migration/v2/import.js

```diff
@@ -0,0 +1,294 @@
+};
+
+const processRow = async (row) => {
+  const productName = row.getCell(1).value;
+  const synonyms = row.getCell(2).value;
+  const combinations = row.getCell(3).value;
+  const routeName = row.getCell(4).value;
+  const doseFormName = row.getCell(5).value;
+  const doseQualificationName = row.getCell(6).value;
+  const doseStrengthName = row.getCell(7).value;
+  const doseUnitName = row.getCell(8).value;
+  const immediatePackagingName = row.getCell(9).value;
+  const packSizeName = row.getCell(10).value;
+  const outerPackagingName = row.getCell(11).value;
+
+  const categoryCode = row.getCell(12).value;
+  const productCode = row.getCell(13).value;
+  const routeCode = row.getCell(14).value;
+  const doseFormCode = row.getCell(15).value;
+  const doseQualificationCode = row.getCell(16).value;
+  const doseStrengthCode = row.getCell(17).value;
+  const doseUnitCode = row.getCell(18).value;
+  const immediatePackagingCode = row.getCell(19).value;
+  const packSizeCode = row.getCell(20).value;
+  const outerPackagingCode = row.getCell(21).value;
+
+  // Properties
+  const gs1 = row.getCell(22).value;
+  const rxNavUC2 = row.getCell(23).value;
+  const rxNavUC67 = row.getCell(24).value;
+  const atcUC2 = row.getCell(25).value;
+  const dddUC2 = row.getCell(26).value;
+  const atcUC5 = row.getCell(27).value;
+  const dddUC5 = row.getCell(28).value;
+  const emlUC2 = row.getCell(29).value;
+  const emlUC67 = row.getCell(30).value;
+  const unspsc = row.getCell(31).value;
+  const nzulmUC2 = row.getCell(32).value;
+  const nzulmUC67 = row.getCell(33).value;
+  const snomed = row.getCell(34).value;
+  const drugbank = row.getCell(35).value;
+  const usfda = row.getCell(36).value;
+
+  const productDefinition = [
+    {
+      type: 'Product',
+      name: productName,
+      code: productCode,
+      properties: [
+        { type: 'code_nzulm', value: nzulmUC2 },
+        { type: 'code_rxnav', value: rxNavUC2 },
+        { type: 'code_unspsc', value: unspsc },
+        { type: 'code_who_atc', value: atcUC2 },
+        { type: 'ddd', value: dddUC2 },
+        { type: 'who_eml', value: emlUC2 },
+      ],
+    },
+    {
+      type: 'Route',
+      name: routeName,
+      code: routeCode,
+      properties: [],
+    },
+    {
+      type: 'DoseForm',
+      name: doseFormName,
+      code: doseFormCode,
+      properties: [],
+    },
+    {
+      type: 'DoseQualification',
+      name: doseQualificationName,
+      code: doseQualificationCode,
+      properties: [
+        { type: 'code_who_atc', value: atcUC5 },
+        { type: 'ddd', value: dddUC5 },
+      ],
+    },
+    {
+      type: 'DoseStrength',
+      name: doseStrengthName,
+      code: doseStrengthCode,
+      properties: [],
+    },
+    {
+      type: 'DoseUnit',
+      name: doseUnitName,
+      code: doseUnitCode,
+      properties: [],
+    },
+  ];
+
+  const [product] = productDefinition;
+
+  if (product.name.indexOf('/') > -1) {
```

Comment:
Think I'm getting an error here:

```
TypeError: Cannot read property 'indexOf' of null
    at processRow (C:\Users\willt\Projects\sussol\unified-codes\unified-codes\tools\scripts\dgraph\migration\v2\import.js:136:22)
    at createImport (C:\Users\willt\Projects\sussol\unified-codes\unified-codes\tools\scripts\dgraph\migration\v2\import.js:239:13)
```

Looks like `productName` is `null`? Could be a spreadsheet issue?

Reply:
I'm getting the same here at the end of the spreadsheet - I think because there are still technically rows in the worksheet, they are just blank. Can probably tidy it up to terminate more tidily when a blank product name is encountered.

Reply:
Yeah, it's no biggie, everything still works, so no need to prioritise here, I don't think?